.github/workflows/ci.yaml

```diff
@@ -1,37 +1,39 @@
 name: CI
 
-on:
-  push:
-    branches: [ master ]
-  pull_request:
-    branches: [ master ]
+on: push
 
 jobs:
   ci:
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.os }}
+    env:
+      STACK_YAML: stack-${{ matrix.ghc }}.yaml
     strategy:
+      fail-fast: false
       matrix:
         ghc: ['8.8']
-        os: [ubuntu-latest, macOS-latest, windows-latest]
+        os: [ubuntu-latest, macOS-latest]
     steps:
     - uses: actions/checkout@v2
     - uses: actions/setup-haskell@v1.1
       with:
         enable-stack: true
         ghc-version: ${{ matrix.ghc }}
         stack-version: '2.3'
+    - name: Setup stack
+      run: |
+        stack config set install-ghc false
+        stack config set system-ghc true
     - name: Cache
       uses: actions/cache@v1
       with:
         path: ~/.stack
-        key: ${{ matrix.os }}-${{ matrix.ghc }}-${{ hashFiles('stack.yaml') }}-${{ hashFiles('**/*.cabal') }}
+        key: a-${{ matrix.os }}-${{ matrix.ghc }}-${{ hashFiles('stack.yaml') }}-${{ hashFiles('**/*.cabal') }}
```

Comment:
Is this just to bust the cache?

Reply:
sadly yes. There is no nicer way (I found so far). I'd love if GH could hash the cache expression itself and use that as a "cache epoch".

Overall I'm also quite unhappy with the GH provided `actions/cache` workflow. I consider to replace it with my own solution.